### PR TITLE
Error out when 0 lines of code were covered

### LINKED ISSUE #488: Zero percent code coverage is not an issue for Infection
| Question    | Answer
| ------------| ---------------
| Infection version | master
| Test Framework version | any

As shown in https://github.com/infection/infection/issues/246#issuecomment-423861590 Infection does not hinder at all, and continue as usual even if there's a zero percent coverage. Instead it would make a perfect sense to abruptly stop with an error. This way a user will immediately know they have a problem with code coverage reporting.

For example, in the case of a coverage report below, it makes no sense for Infection to continue since all mutants will be uncovered.
```                         
 Summary:                
  Classes:  0.00% (0/12) 
  Methods:  0.00% (0/27) 
  Lines:    0.00% (0/152)
```
```
Processing source code files: 19/19
Creating mutated files and processes: 147/147
.: killed, M: escaped, S: uncovered, E: fatal error, T: timed out

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS   ( 50 / 147)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS   (100 / 147)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS      (147 / 147)

147 mutations were generated:
       0 mutants were killed
     147 mutants were not covered by tests
       0 covered mutants were not detected
       0 errors were encountered
       0 time outs were encountered
```
The only case where Infection can continue without an error is when a user explitedly set required minimal MSI to zero. So we could understand that the user knows what they doing.

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Make infection error when presented with 0 lines of executed code
- [x] Covered by tests

Fixes #488

### CODE REVIEW

File: src/TestFramework/PhpUnit/Coverage/CoverageXmlParser.php [hunk 1]
```diff
@@ -74,6 +77,17 @@ public function parse(string $coverageXmlContent): array
...
     }
 
+    private function assertHasCoverage(\DOMXPath $xPath): void
+    {
+        $lineCoverage = $xPath->query('/phpunit/project/directory[@name="/"]/totals/lines');
+
+        Assert::notSame(
+            '0',
+            $lineCoverage->item(0)->getAttribute('executed'),
+            '0 Lines of code were executed during tests. Please check your filters.'
```
Comment (borNfreee) [inline src/TestFramework/PhpUnit/Coverage/CoverageXmlParser.php]: What are the `filters`? I guess the end user will have the same question. Could you please change the exception message to something more descriptive:

* like filters of Infection or PHPUnit
* possible ways of debugging the issue (enable `--debug`?, looking at the `infection.log`? etc.). Anything that can help

**Comment by borNfreee** [comment]
@BackEndTea could you please run `make cs-fix`?

**Comment by borNfreee** [comment]
PHPStan fails with the error:

```
 ------ ---------------------------------------------------------- 
  Line   src/TestFramework/PhpUnit/Coverage/CoverageXmlParser.php  
 ------ ---------------------------------------------------------- 
  86     Cannot call method getAttribute() on DOMNode|null.        
 ------ ----------------------------------------------------------
```

**Comment by borNfreee** [comment]
Builds are failing:

* https://ci.appveyor.com/project/borNfreee/infection/builds/23985790/job/c9tu7bgynpujtp33#L49009
* https://travis-ci.org/infection/infection/jobs/522419483#L693

**Comment by BackEndTea** [comment]
I can't really reproduce this error, it appears to be windows (appveyor) only. And only on php 7.1

**Comment by maks-rafalko** [comment]
Thank you @BackEndTea, finally it's in master 👍 💃

**Pull request merged**